diff --git a/node_modules/medium-zoom/dist/pure/index.js b/node_modules/medium-zoom/dist/pure/index.js
index d4ca61a..227fb90 100644
--- a/node_modules/medium-zoom/dist/pure/index.js
+++ b/node_modules/medium-zoom/dist/pure/index.js
@@ -14,7 +14,7 @@ var _extends = Object.assign || function (target) {
 };
 
 var isSupported = function isSupported(node) {
-  return node.tagName === 'IMG';
+  return node.tagName === 'IMG' || node.tagName === 'SVG';
 };
 
 /* eslint-disable-next-line no-prototype-builtins */
@@ -28,7 +28,7 @@ var isNode = function isNode(selector) {
 
 var isSvg = function isSvg(image) {
   var source = image.currentSrc || image.src;
-  return source.substr(-4).toLowerCase() === '.svg';
+  return typeof source === 'string' && source.substr(-4).toLowerCase() === '.svg';
 };
 
 var getImagesFromSelector = function getImagesFromSelector(selector) {
@@ -72,7 +72,7 @@ var cloneTarget = function cloneTarget(template) {
       width = _template$getBounding.width,
       height = _template$getBounding.height;
 
-  var clone = template.cloneNode();
+  var clone = template.tagName === 'SVG' ? template.cloneNode(true) : template.cloneNode();
   var scrollTop = window.pageYOffset || document.documentElement.scrollTop || document.body.scrollTop || 0;
   var scrollLeft = window.pageXOffset || document.documentElement.scrollLeft || document.body.scrollLeft || 0;
 
